### ISSUE TITLE
XGBoost model returns numeric codes instead of crop names due to missing LabelEncoder persistence

### ISSUE THREAD

**Issue by shupandee** [vannu07/Farm-IQ-AI-Powered-Smart-Farming-Assistant#14]
Description
The XGBoost model currently uses LabelEncoder to convert crop names (string labels) into integers during training.
However, only the trained model is saved to disk — the LabelEncoder itself is not persisted.
As a result, when the model is reloaded for inference, it outputs encoded integers (e.g., 0–21) instead of human-readable crop names.
Impact
Severity: High
Breaks production deployment
Makes the trained XGBoost model unusable without retraining
Affects API endpoints depending on decoded crop names
Farmers receive numeric codes instead of actual crop names
Expected Behavior
Both the trained model and the LabelEncoder should be saved and reloaded together so predictions can be decoded back to original crop names.
Proposed Solution
Save the model and encoder together in one file (or separately if preferred).
Example implementation:
import joblib
# During training
model.fit(X_train, y_train)
joblib.dump({'model': model, 'encoder': label_encoder}, 'crop_model.pkl')
# During inference
data = joblib.load('crop_model.pkl')
model = data['model']
encoder = data['encoder']
decoded_preds = encoder.inverse_transform(model.predict(X_test))
Additional Context
This issue impacts model deployment and user-facing outputs.
Please consider adding the hacktoberfest label.
I’d also like to work on fixing this issue myself. 🙌

**Comment by vaibhavjais2503** [comment]
Hi! I’m interested in fixing this for Hacktoberfest.

Plan:
- Persist LabelEncoder with joblib during training
- Load encoder + model at inference and return crop names (not codes)
- Add a small pytest for inverse_transform round-trip
- Update API docs if response shape changes

If the current assignee is still on it, no worries—I can collaborate or take a related sub-task (e.g., tests + docs). Otherwise, could you please assign me?

cc: @vannu07 @shupandee

**Comment by shupandee** [comment]
@vannu07 hi please see the pr request
